add total ad click and total search with ads measures

mmccorks opened the pull request:
Checklist for reviewer:

When adding a new derived dataset:
- [ ] Ensure that the data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data may be available in [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).
- [ ] Avoid merging a PR that includes the logic of a [core metric](https://docs.telemetry.mozilla.org/metrics/index.html) or complex business logic. The recommendation is to implement core business logic in bigquery-etl. E.g. The [type of search](https://github.com/mozilla/bigquery-etl/blob/a3e59f90326816a2ecaaa3e9d5b57fe9552f7d70/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_daily_v1/query.sql#L781) or the [calculation of DAU or visited URIs](https://github.com/mozilla/bigquery-etl/blob/9bca48821a8a0d40b1700cc14ecd8068d132ed06/sql/moz-fx-data-shared-prod/telemetry_derived/firefox_desktop_exact_mau28_by_dimensions_v1/query.sql).
- [ ] Avoid merging code in Looker Explores/Views that implement analysis with multiple lines of code or that will be likely replicated in the future. Instead, aim for extending an existing dataset to include the required logic, and use [Looker aggregates](https://cloud.google.com/looker/docs/aggregate_awareness) to facilitate the analysis.
- [ ] Avoid merging a PR with logic that requires validation and health checks. It is recommended to implement it in bigquery-etl for full test coverage and failure alerts.

ANich commented on firefox_desktop/views/newtab_interactions.view.lkml:
```diff
@@ -195,6 +195,20 @@ view: newtab_interactions {
+    group_label: "Search"
+    label: "Total Ad Clicks"
+    type: sum
+    sql: ${tagged_search_ad_clicks} + ${follow_on_search_ad_clicks} ;;
```
I don't think these are mutually exclusive - see above `sum_tagged_follow_on_search_ad_clicks`, it's defined AS:

```
  COUNTIF(
    is_tagged_search_ad_click
    AND is_follow_on_search_ad_click
  ) AS tagged_follow_on_search_ad_clicks,
```
Same for ad impressions

mmccorks replied:
Ah ok, so this means that follow on is a subset of tagged searches then. This is different than our usual way we define these things, let me make the appropriate adjustments.

mmccorks replied:
I've made some edits, can you take a look @ANich? 
Do we need to update the dimensions definitions in the view as well? And should we add labels to all of these metrics?

ANich replied:
See https://github.com/mozilla/looker-spoke-default/pull/452#issuecomment-1325227620, this should probably be updated at the source so it doesn't cause confusion elsewhere. Yeah this could definitely use some descriptions

ANich commented:
We should change this upstream, likely here: 

https://github.com/mozilla/bigquery-etl/blob/0647d3ed7a70148eec38a4cfd993914105f5f835/sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v1/query.sql#L160-L161 

So if I understand right, tagged searches is for searches that are tagged but NOT follow-on?

mmccorks commented:
Yes, typically in other tables we have `tagged_sap` and `tagged_follow_on` as separate fields. We also generally call ad impressions `search_with_ads` but I think calling it ad impression here is ok to stay consistent with the tile impression data.

mmccorks commented:
Ok so - I've changed the search definitions so that tagged searches are just tagged searches not including follow-ons to remain consistent with other search tables, added total tagged search which contains both tagged and tagged follow on.
I removed untagged follow on searches with ads/untagged follow on ad clicks as these should not be any different than tagged follow on (we do not have follow on searches with ads or ad clicks instrumented for non-partner search engines) and to reduce confusion.
